Modules/TPC/src/TrendingTaskTPC.cxx

```diff
@@ -289,22 +298,30 @@ void TrendingTaskTPC::drawCanvasMO(TCanvas* thisCanvas, const std::string& var,
   // Setup the tree reader with the needed values.
   TTreeReader myReader(mTrend.get());
   TTreeReaderValue<UInt_t> retrieveTime(myReader, "time");
+  TTreeReaderValue<Long64_t> retrieveRun(myReader, "meta.runNumber");
   TTreeReaderValue<std::vector<SliceInfo>> dataRetrieveVector(myReader, varName.data());
 
   const int nuPa = mNumberPads[varName];
   const int nEntries = mTrend->GetEntriesFast();
+  const int nEntriesTime = mTrend->GetBranch("time")->GetEntries();
+  const int nEntriesRuns = mTrend->GetBranch("meta")->GetEntries();
+  const int nEntriesData = mTrend->GetBranch(varName.data())->GetEntries();
 
   // Fill the graph(errors) to be published.
-  if (trendType == "time") {
+  if (trendType == "time" || trendType == "run") {
+    const int nEffectiveEntries = (trendType == "time") ? std::min(nEntriesTime, nEntriesData) : std::min(nEntriesRuns, nEntriesData);
+    const int startPoint = (trendType == "time") ? nEntriesTime - nEffectiveEntries : nEntriesRuns - nEffectiveEntries;
 
     for (int p = 0; p < nuPa; p++) {
       thisCanvas->cd(p + 1);
       int iEntry = 0;
 
-      graphErrors = new TGraphErrors(nEntries);
+      graphErrors = new TGraphErrors(nEffectiveEntries);
+      myReader.SetEntry(startPoint - 1); // startPoint-1 as myReader.Next() increments by one so that we then start at startPoint
 
       while (myReader.Next()) {
-        const double timeStamp = (double)(*retrieveTime);
+        const double timeStamp = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);
+        ;
```

Comment:
There is an empty line with `;`. And perhaps for better readability `timeStamp` -> `xVal`
```suggestion
        const double xVal = (trendType == "time") ? (double)(*retrieveTime) : (double)(*retrieveRun);

```